### ISSUE TITLE
Possible to support 480p?

### ISSUE THREAD

**Issue by seanlaguna** [tolga9009/elgato-gchd#17]
Hey,

I haven't delved into the code much, but would it be hard to support 480p or 480i? I'm thinking of using this to record smash bros melee from a wii with component cables. How much work would be involved? If it's not much and you give me some direction, I may be able to do it myself.

Thanks!

**Comment by tolga9009** [comment]
Hi there,

this is on the TODO list (NTSC support). I'm based in Europe and just had access to PAL devices, that's why it hasn't been implemented yet, I'm sorry for that. However, someone told me you can switch a hacked original Xbox from PAL to NTSC. Luckily, I have a softmodded original Xbox around and will use it for adding NTSC support.

Without going in to much detail, I can easily say that reverse-engineering this device is non-trivial. Including NTSC support. Problem is, you can't easily capture USB traffic, cause video capturing / encoding is a very CPU intensive task, thus making USB traffic capturing in a virtual machine almost impossible.

Maybe it's possible in the meantime (using GPU / PCIe passthrough, paravirtualization etc.), but last time I've checked (Intel Xeon E3-1245v2, 16GB RAM, HD7970), the VM lagged and the captures wer incomplete / faulty.

But even if VMs worked, we would need to port the reverse-engineering tools, which are currently firmly tailored to a workflow using the OpenVizsla USB hardware analyzer.

**TL;DR** you will need an external hardware USB protocol analyzer and 2 PCs for monitoring / capturing USB traffic.

I'm currently busy with another project, but will pick this one up again. I have everything here to reverse-engineer the NTSC support and it's definitely on my TODO list. But please don't hold your breath on it, I can't name any release date.

Cheers,
Tolga

**Comment by StianOby** [comment]
Changing from PAL to NTSC on a softmodded Xbox is very easy, yes. Install the program "Enigmah videomode switch disk" and run it. You will then be presented with this screen, where you can choose region: http://xbox.douglasstridsberg.com/wp-content/uploads/switchdisc.jpg

(We Halo players use this a lot to play the NTSC version of the original Halo: Combat Evolved. This is because the PAL version is terrible and slow, since all animations were frame-based. Bungie didn't realize the mistake until it was too late to fix...)

**Comment by NerdyShawn** [comment]
+1 on the 480p for me as well, I have a NTSC gamecube with the component cables as well interested in testing. So far great work though going to play around with this, hoping to do some Twitch Streaming but don't want to build a windows PC just for that. (i'm doubting windows vm on my i7 system would work well, despite if I gave it plenty of resources). Good works, thanks again.

**Comment by tolga9009** [comment]
Hi there! I have a modded Xbox and it shouldn't be a problem to get NTSC working on it. However, I don't have a component cable, so the best I can get is 480i for you. I will work on this now.

**Comment by tolga9009** [comment]
After reading a bit, I found out PAL PS2s are supporting 480i and 480p, as long as the game supports it. I should have a couple of NTSC games around for 480i. I also happen to have a component cable for PS2 and Tekken 4, which outputs 480p.

I'll keep you up-to-date about any further developments.

**Comment by StianOby** [comment]
@tolga9009 : I have a spare set of component cables (YPbPr) for the original Xbox. If you want, I can mail them to you. (You live in Germany, right? If so, it shouldn't cost me much.) Just send me your address on a Gitter PM.

**Comment by seanlaguna** [comment]
I could also mail you some component cables if you want, I would be happy to buy them to support teh cause.

My use case is also a NSTC (N. Am.) gamecube for playing/streaming smash bros melee on linux. :)

**Comment by tolga9009** [comment]
Thanks for the support, but I was able to get a 480i and 480p signal from my PS2. I'm gonna implement the following setups:

1. PS2 NTSC 480i, 60 FPS -> Composite Cable -> Elgato -> HDMI Out -> Full HD Monitor
2. PS2 NTSC 480p, 60 FPS -> Component Cable -> Elgato -> HDMI Out -> Full HD Monitor

If your setups are different from mine, please tell me. The implementations should be available in the upcoming few hours.

**Comment by tolga9009** [comment]
Support has been added for 480i (b400e90e0689ce96c0700dd5e1afa1f577a34570) and 480p (d97aa003f7861b3fac0a22dee00d413a4ba2b5e7). You can try it out using `gchd -r ntsc -i composite` for 480i and `gchd -r ntsc -i component` for 480p.

Please note, that overall usage has changed between `master` and `refactoring` branch, so please wait until the README has been updated. Feel free to reopen, if this issue hasn't been fixed for you.

Thanks for reporting! Closing now.
